### DISCUSSION TITLE
Managed Refinable Properties from Active Directory

### DISCUSSION THREAD

**Discussion by charlespaoletti** [microsoft-search/pnp-modern-search#3200] [Q&A]
Hello,

I am trying to create an employee directory using the instruction [here](https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-useful-People-Search/). It is mostly working but some of the People properties arent available for filtering, for example Department is not available. Job title was in the list of selectable properties for filtering and complete contact cards appear on hover, so I know some properties are coming to SharePoint from Active Directory. 

I've found articles and discussions about adding "Managed Refinable" properties from lists in SharePoint but not how to do that with existing user properties from AD.

How do I get Department working and how do I add other user properties from AD like Office location and Employee type so that I can add them as Filters?

Thanks so much!

**Comment by kasperbolarsen** [comment]
Hi,
I would suggest starting with verifying you User Profile data using the SharePoint Admin portal. Once you are sure that at least one person has a value in the Department field in the User profile, I would verify the data in the Search index using SP Query Tool, see https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/
If the data in in the User Profile and not in the search index, you might have to reindex the User Profile data, see https://www.techmikael.com/2014/12/how-to-trigger-re-indexing-of-user.html

:-)

**Reply by charlespaoletti** [reply]
Thanks for the reply. That is definitely a usefule tool!

So I am really confused now. Because neither Department OR Job Title is in the returned list from the Query Tool, however Job Title is available as a filter(and it works correctly). Both properties have values in multiple User Profiles. They are also listed in the Term Store and they seem to have gotten the relevant terms directly from AD because I did not manually enter them.

They are both listed in Managed Properties but they have drastically different values and settings.

Thanks for the help.

**Reply by kasperbolarsen** [reply]
are you using the Local People Results source ?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/ca046ae8-2d8d-42db-ba02-5cf6df2252d9)

**Comment by charlespaoletti** [comment]
No I wasnt. Now I get 0 results.

I put my url in SharePoint Site URL, Sign in with Admin account, Query Text us "sharepoint", Refiners is  "managedproperties(filter=600/0/*)" and Result source Id is Local People Results.

**Comment by kasperbolarsen** [comment]
I would recommend starting with the most basic setup:
Set ResultSource id to People Result, set the Query Text to either * or your first name, and take a look :
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/4e55b646-2466-4d03-a76c-6732d4ac45ad)

**Reply by charlespaoletti** [reply]
When I do that all the users show a Department, and Job Title. But Department does not show as an available Filter Field.

**Comment by kasperbolarsen** [comment]
Excellent, the data is there :-) Have you added Department and JobTitle to the "Selected Properties" in the Results Web Part? You MUST have added the field to the Selected Properties in the Results Web part in order to use them in the Refiner Web part.

**Comment by charlespaoletti** [comment]
I hadnt even looked at that yet. Very new to PnP Modern Search. Job Title and a bunch of other properties were selected by default. But Department is not a chooice in the selected propertied list.

**Comment by kasperbolarsen** [comment]
We have all been new to PnP Modern Search and sometime it takes time to learn how to connect the dots :-)

Did that solve your problem? If now you could request one of the slots available in my PnP Modern Search Office Hours ( https://ms365thinking.blogspot.com/2023/08/pnp-modern-search-office-hours-online.html) , it free and I promise I will not bite ;-)

**Comment by charlespaoletti** [comment]
No. Department is still not in the filter list or the selectable properties. And I screwed something up becasue Job Title is not working anymore. I will take a look at your office hours. Thanks so much. I submitted a request for September 5th.
